eng/packages/http-client-csharp/doc/DEV_ONBOARDING.md

```diff
@@ -0,0 +1,232 @@
+- **Use Case**: Non-Azure services or generic HTTP APIs
+- **Package.json Artifact**: `eng/http-client-csharp-emitter-package.json`
+
+### 2. Azure Data Plane Emitter (`@azure-typespec/http-client-csharp`)
+
+- **Purpose**: Generates Azure-branded data plane client libraries
+- **Repository**: [Azure/azure-sdk-for-net](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/packages/http-client-csharp)
+- **Use Case**: Azure data plane services (e.g., Azure Storage, Cognitive Services)
+- **Package.json Artifact**: `eng/azure-typespec-http-client-csharp-emitter-package.json`
+
+### 3. Azure Management Plane Emitter (`@azure-typespec/http-client-csharp-mgmt`)
+
+- **Purpose**: Generates Azure Resource Manager (ARM) client libraries
+- **Repository**: [Azure/azure-sdk-for-net](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/packages/http-client-csharp-mgmt)
+- **Use Case**: Azure management plane services (ARM resources)
+- **Package.json Artifact**: `eng/azure-typespec-http-client-csharp-mgmt-emitter-package.json`
+
+## Key Concepts
+
+### tsp-client
+
+**tsp-client** is the orchestration tool that partners and build pipelines use to invoke the TypeSpec compiler and emitters.
+
+- **Repository**: [Azure/azure-sdk-tools](https://github.com/Azure/azure-sdk-tools/tree/main/tools/tsp-client)
+- **Key Behavior**: Expects a corresponding `package.json` file to install the correct emitter and compile TypeSpec definitions
+- **Integration**: Automatically invoked by the .NET SDK build system when generating code
+
+### Configuration Files
+
+#### tspconfig.yaml
+
+Defines the TypeSpec compilation settings and emitter configuration.
+
+**Required Configuration for .NET Generation:**
+```yaml
+options:
+  "@azure-typespec/http-client-csharp":
+    emitter-output-dir: "{output-dir}/{service-dir}/{namespace}"
+    namespace: Azure.Data.AppConfiguration
+    model-namespace: false
+```
+
+**Key Options:**
+- `emitter-output-dir`: Directory where generated code will be placed
+- `namespace`: The .NET namespace for the generated client
+- `model-namespace`: Whether to use a separate namespace for models (typically `false`)
+
+#### tsp-location.yaml
+
+Maps the library to its TypeSpec specification and emitter configuration.
+
+**Key Properties:**
+- `directory`: Path to the TypeSpec specification (typically in azure-rest-api-specs repo)
+- `commit`: Git commit SHA of the spec version to use
+- `repo`: Repository containing the TypeSpec specification
+- `emitterPackageJsonPath`: Path to the emitter package.json artifact
+
+**Example:**
+```yaml
+directory: specification/cognitiveservices/OpenAI.Inference
+commit: abc123def456
+repo: Azure/azure-rest-api-specs
+emitterPackageJsonPath: eng/azure-typespec-http-client-csharp-emitter-package.json
+```
+
+#### Library .csproj Configuration
+
+To exclude legacy AutoRest dependencies, add:
+
+```xml
+<PropertyGroup>
+  <IncludeAutorestDependency>false</IncludeAutorestDependency>
```

Comment:
We probably won't need this once the other PR merges that switches the default behavior.